source/Irrlicht/CNullDriver.h

```diff
@@ -53,8 +53,11 @@ namespace video
 
         bool bindGraphicsPipeline(video::IGPURenderpassIndependentPipeline* _gpipeline) override { return false; }
 
-        bool bindDescriptorSets(E_PIPELINE_BIND_POINT _pipelineType, uint32_t _first, uint32_t _count,
-            video::IGPUDescriptorSet** _descSets, uint32_t _dynOffsetCount, const uint32_t* _dynOffsets) override { return false; }
+        bool bindDescriptorSets(E_PIPELINE_BIND_POINT _pipelineType, const IGPUPipelineLayout* _layout,
+            uint32_t _first, uint32_t _count, const IGPUDescriptorSet** _descSets, core::smart_refctd_dynamic_array<uint32_t>* _dynamicOffsets) override 
+        { 
+            return false; 

```

Comment:
I would put the generic tracking here.

Reply:
seriously.

Reply:
that would require dividing GL state tracking (which is per-context) into COpenGLDriver and CNullDriver.. Practically undoable

Reply:
ok, but can you put some of the code (as a static protected function) or the state tracking object into CNullDriver namespace?

Reply:
basically the code that decides whether to throw out a descriptor set is pretty generic and will be needed in Vulkan, would be sad to have it buried deep in COpenGLDriver

Reply:
resolved

Reply:
resolved

Reply:
resolved